src/DeepCopy/DeepCopy.php

```diff
@@ -40,35 +39,43 @@
      */
     private $typeFilters = [];
 
-    private $skipUncloneable = false;
+    private $skipUncloneable;
     private $useCloneMethod;
 
     /**
-     * @param bool $useCloneMethod If set to true, when an object implements the __clone() function, it will be used
-     *                             instead of the regular deep cloning.
+     * @param bool $useCloneMethod           If set to true, when an object implements the __clone() function, it will
+     *                                       be used instead of the regular deep cloning.
+     * @param bool $skipUncloneable          If enabled, will not throw an exception when coming across an uncloneable
+     *                                       property.
+     * @param Array<Filter, Matcher>         List of filter-matcher pairs
+     * @param Array<TypeFilter, TypeMatcher> List of type filter-matcher pairs
      */
-    public function __construct(bool $useCloneMethod = false)
-    {
+    public function __construct(
+        bool $useCloneMethod = false,
+        bool $skipUncloneable = false,
+        array $filters = [],
+        array $typeFilters = []
```

Comment:
Maybe we can upgrade this typehint to [iterable](https://secure.php.net/manual/en/language.types.iterable.php) and in deep_copy function too.